cron/controller/main.go

```diff
@@ -99,18 +102,37 @@ func main() {
 		panic(err)
 	}
 
-	shardNum, err := publishToRepoRequestTopic(ctx, reader, t)
+	bucket, err := config.GetResultDataBucketURL()
 	if err != nil {
 		panic(err)
 	}
-	bucket, err := config.GetResultDataBucketURL()
+
+	shardNum, err := publishToRepoRequestTopic(ctx, reader, t)
 	if err != nil {
 		panic(err)
 	}
+	// TODO(azeems): Stop populating `.shard_num` file.
 	err = data.WriteToBlobStore(ctx, bucket,
 		data.GetShardNumFilename(t),
 		[]byte(strconv.Itoa(int(shardNum+1))))
 	if err != nil {
 		panic(err)
 	}
+	// Populate `.shard_metadata` file.
+	metadata := data.ShardMetadata{
+		NumShard:  new(int32),
+		ShardLoc:  new(string),
+		CommitSha: new(string),
+	}
+	*metadata.NumShard = (shardNum + 1)
+	*metadata.ShardLoc = bucket + "/" + data.GetBlobFilename("", t)
+	*metadata.CommitSha = os.Getenv(commitSHA)
+	metadataJSON, err := protojson.Marshal(&metadata)
+	if err != nil {
+		panic(err)
```

Comment:
maybe add error message in these two panics ?

Reply:
Done.